scene/main/viewport.cpp

```diff
@@ -1985,13 +1985,16 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 
 		if (gui.drag_data.get_type() == Variant::NIL && over && !gui.modal_stack.empty()) {
 
+			//  Dragging nodes causes re-search of currently moused-over control
+			Control *dragged = _gui_find_control(mpos);
+
 			Control *top = gui.modal_stack.back()->get();
 
-			if (over != top && !top->is_a_parent_of(over)) {
+			if (dragged != top && !top->is_a_parent_of(dragged)) {
```

Comment:
Not using `over` here and recalling `_gui_find_control` seems to defeat the purpose of lines 1977-1984 which computer `over` and cache its value to avoid rechecking the stack all the time.

So I'm not sure this is correct, it seems a bandaid and not a complete fix yet.